ooniprobe/View/RunTestViewController.mm

```diff
@@ -0,0 +1,121 @@
+    [self.test_iconImage setImage:[UIImage imageNamed:[NSString stringWithFormat:@"%@_big", current.name]]];
+    
+    self.tableView.estimatedRowHeight = 44.0;
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+
+    urls = [testArguments objectForKey:@"urls"];
+    if ([urls count] > 0){
+        showIcon = TRUE;
+        rows = urls;
+    }
+    else if ([current.name isEqualToString:@"web_connectivity"]){
+        rows = [NSArray arrayWithObject:NSLocalizedString(@"random_sampling_urls", nil)];
+    }
+    else {
+        [self.tableView setSeparatorStyle:UITableViewCellSeparatorStyleNone];
+    }
+    
+    [self reloadTestUI];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+-(IBAction)close:(id)sender{
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+-(void)reloadTestUI{
+    NetworkMeasurement *current = [currentTests getTestWithName:testName];
+    //basic case (not running e not started)
+    if (!current.running){
+        [self.indicator setHidden:TRUE];
+        [self.runButton setHidden:FALSE];
+        [self.indicator stopAnimating];
+    }
+    //current test is running
+    else {
+        [self.indicator setHidden:FALSE];
+        [self.runButton setHidden:TRUE];
+        [self.indicator startAnimating];
+    }
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return [rows count];
+}
+
+- (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section
+{
+    if ([rows count] == 0)
+        return nil;
+    return NSLocalizedString(@"urls", nil);
+}
+
+- (void)tableView:(UITableView *)tableView willDisplayHeaderView:(UIView *)view forSection:(NSInteger)section{
+    UITableViewHeaderFooterView *header = (UITableViewHeaderFooterView *)view;
+    header.backgroundView.backgroundColor = [UIColor clearColor];
+    header.textLabel.font = [UIFont fontWithName:@"FiraSansOT-Bold" size:18];
+    [header.textLabel setTextColor:color_off_black];
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
+    NSString *current = [rows objectAtIndex:indexPath.row];
+    if (showIcon) cell.imageView.image = [UIImage imageNamed:@"include_cc"];
+    cell.textLabel.text = current;
+    return cell;
+}
+
+-(IBAction)run:(id)sender {
+    NetworkMeasurement *current = [currentTests getTestWithName:testName];
+    if ([urls count] > 0){
+        current.inputs = urls;
+    }
+    [current run];
+    [self dismissViewControllerAnimated:TRUE completion:^{
+        [[NSNotificationCenter defaultCenter] postNotificationName:@"reloadTable" object:nil];
```

Comment:
So, does this mean that, when the view is closed, you post the `@"reloadTable"` message that is the one you use to reload, if I understand correctly, the `run test` view and, possibly, also the `past test results` view?

Reply:
Is the rationale of posting this message that you will then be able to update the other views since you basically have started the test from another code-path and view?

Reply:
Yes, because I will post this message after dismissing this view controller

Reply:
Okay, thanks